lib/tasks/data_requests.rake

```diff
@@ -0,0 +1,56 @@
+namespace :data_requests do
+  desc 'Recursively lookup users using a network of shared devices'
+  task lookup_users_by_device: :environment do
+    ActiveRecord::Base.connection.execute('SET statement_timeout = 0')
+
+    uuids = ENV.fetch('UUIDS', '').split(',')
+    users = uuids.map { |uuid| DataRequests::LookupUserByUuid.new(uuid).call }.compact
+
+    result = DataRequests::LookupSharedDeviceUsers.new(users).call
+    puts JSON.pretty_generate(result)
+    puts "UUIDS: #{result.values.flatten.uniq.join(',')}"
+  end
+  # UUIDS=123abc,456def rake data_requests:lookup_users_by_device
+
+  desc 'Create a JSON report with data for the specified users'
+  task create_users_report: :environment do
+    uuids = ENV.fetch('UUIDS', '').split(',')
+    requesting_issuers = ENV.fetch('REQESTING_ISSUER', nil)&.split(',')
+
+    output = uuids.map do |uuid|
+      user = DataRequests::LookupUserByUuid.new(uuid).call
+      next warn("No record for uuid #{uuid}") if user.nil?
+      DataRequests::CreateUserReport.new(user, requesting_issuers).call
+    end.compact.to_json
+    puts output
+  end
+  # UUIDS=123abc,456def REQUESTING_ISSUER=sample:app:issuer rake data_requests:create_users_report
```

Comment:
Does rake read these comments below? I'd expect them to go above the tasks

Reply:
I copied this pattern from another rake task I was looking at to remind myself what the syntax is: https://github.com/18F/identity-idp/blob/master/lib/tasks/deleted_accounts.rake#L13

But, am happy to move the comments to the top.